### PR TITLE
Simple auth manager documentation

### PR DESCRIPTION
**Pull request by vincbeck**
Create documentation for simple auth manager.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by shahar1** [comment]
Fails in static checks

File: docs/apache-airflow/core-concepts/auth-manager/index.rst [hunk 1]
```diff
@@ -37,6 +37,14 @@ If you want to check which auth manager is currently set, you can use the
     $ airflow config get-value core auth_manager
     airflow.providers.fab.auth_manager.fab_auth_manager.FabAuthManager
 
+Available auth managers to use
+------------------------------
+
+Here is the list of auth managers available today you can use in your Airflow environment.
```
Comment (o-nikolas) [inline docs/apache-airflow/core-concepts/auth-manager/index.rst]:
```suggestion
Here is the list of auth managers available today that you can use in your Airflow environment.
```

File: docs/apache-airflow/core-concepts/auth-manager/simple.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+Simple auth manager
+===================
+
+.. note::
+    Before reading this, you should be familiar with the concept of auth manager.
+    See :doc:`apache-airflow:core-concepts/auth-manager/index`.
+
+.. warning::
+  The simple auth manager is intended to be used for development and testing purposes. It should not be used in production environment.
```
Comment (o-nikolas) [inline docs/apache-airflow/core-concepts/auth-manager/simple.rst]:
```suggestion
  The simple auth manager is intended to be used for development and testing purposes. It should not be used in a production environment.
```

File: docs/apache-airflow/core-concepts/auth-manager/simple.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          "role": "admin",
+      }
+  ]
+
+Each user needs two pieces of information:
+
+* **username**. The user's username
+* **role**. The role associated to the user. For more information about these roles, :ref:`see next section <roles-permissions>`.
+
+The password is auto-generated for each users and printed out in the webserver logs.
```
Comment (o-nikolas) [inline docs/apache-airflow/core-concepts/auth-manager/simple.rst]:
```suggestion
The password is auto-generated for each user and printed out in the webserver logs.
```

**Comment by vincbeck** [comment]
It seems modifying docs from Airflow core and providers at the same time is challenging for the CI. It is complaining some document do not exist but they do (in the PR). Therefore, I created a PR for Airflow core only: #42418. Once this one is merged, I'll create another one for providers

**Review by o-nikolas — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]

**Pull request merged**